src/FilenameParsing/FileIDHelperResolutionStrategy.php

```diff
@@ -436,7 +436,7 @@ public function findVariants($tuple, Filesystem $filesystem)
 
             // Find the correct folder to search for possible variants in
             $folder = $helper->lookForVariantIn($parsedFileID);
-            $possibleVariants = $filesystem->listContents($folder, true);
+            $possibleVariants = $filesystem->listContents($folder, $helper->lookForVariantRecursive());
```

Comment:
Prevent it from being a potentially breaking change
```
$recursive = method_exists($helper, 'lookForVariantRecursive') ? $helper->lookForVariantRecursive() : true;
$possibleVariants = $filesystem->listContents($folder, $recursive);
```

Reply:
Even if we do this change, PHP will throw a fit if you have a class that implement the `FileIDHelper` interface without the `lookForVariantRecursive` method.

I would rather keep the API for FileIDHelper clean and concise ... even if it causes an internal API breakage.